File: app/src/main/java/org/autojs/autojs/ui/settings/CheckForUpdatesPreference.java
package org.autojs.autojs.ui.settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.Preference;
import android.util.AttributeSet;

import androidx.preference.PreferenceManager;

import org.autojs.autojs.Pref;
import org.autojs.autojs6.R;

/**
 * Created by SuperMonster003 on May 31, 2022.
 */

public class CheckForUpdatesPreference extends Preference implements SharedPreferences.OnSharedPreferenceChangeListener {
    public CheckForUpdatesPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    public CheckForUpdatesPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        SharedPreferences mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(getContext());
        mSharedPreferences.registerOnSharedPreferenceChangeListener(this);
    }

    public CheckForUpdatesPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public CheckForUpdatesPreference(Context context) {
        super(context);
        init();
    }

    @Override
    protected void onAttachedToActivity() {
        setSummaryIfNeeded();
        super.onAttachedToActivity();
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        setSummaryIfNeeded();
    }

    private void setSummaryIfNeeded() {
        String lastChecked = Pref.getLastUpdatesCheckedTimeString();
        if (lastChecked != null) {
            setSummary(getContext().getString(R.string.text_last_updates_checked_time, lastChecked));
        }
    }
}

File: app/src/main/java/org/autojs/autojs/ui/settings/CheckForUpdatesPreference.kt
package org.autojs.autojs.ui.settings

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.util.AttributeSet
import org.autojs.autojs.core.pref.Pref
import org.autojs.autojs.network.UpdateChecker
import org.autojs.autojs.network.UpdateChecker.PromptMode
import org.autojs.autojs.theme.preference.MaterialPreference
import org.autojs.autojs6.R

/**
 * Created by SuperMonster003 on May 31, 2022.
 */
class CheckForUpdatesPreference : MaterialPreference, OnSharedPreferenceChangeListener {

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context) : super(context)

    init {
        Pref.registerOnSharedPreferenceChangeListener(this)
        summaryProvider = SummaryProvider<CheckForUpdatesPreference> {
            Pref.lastUpdatesCheckedTimeString?.let { prefContext.getString(R.string.text_last_updates_checked_time, it) }
        }
    }

    override fun onClick() {
        UpdateChecker.Builder(prefContext)
            .setPromptMode(PromptMode.DIALOG)
            .build().checkNow()
        super.onClick()
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) = notifyChanged()

}